server/graphql/v2/input/NewPaymentMethodInput.js

```diff
@@ -0,0 +1,14 @@
+import { GraphQLInputObjectType, GraphQLNonNull, GraphQLString } from 'graphql';
+
+import { NewPaymentMethodDataInput } from './NewPaymentMethodDataInput';
+
+export const NewPaymentMethodInput = new GraphQLInputObjectType({
+  name: 'NewPaymentMethodInput',
+  fields: () => ({
+    data: { type: new GraphQLNonNull(NewPaymentMethodDataInput) },
+    name: { type: new GraphQLNonNull(GraphQLString) },
+    service: { type: new GraphQLNonNull(GraphQLString) },
+    token: { type: new GraphQLNonNull(GraphQLString) },
```

Comment:
Should be optional if we want to support non-credit-card payment methods.

If we detect the credit card type, we should then validate that in the mutation itself.